### ISSUE TITLE
[13.0]pos_order_mgmt: Reprint ticket doesent print original paymentlines

### ISSUE THREAD

**Issue by javierjcf** [OCA/pos#732]
## Module
pos_order_mgmt

## Describe the bug
When try to reprint a ticket, i get the ticket without the original paymentlines.

## To Reproduce
Use the reprint ticket button on the order history screen and the payment lines is not printed

**Expected behavior**
A exact copy of the original ticket

**Additional context**
The funtion _prepare_order_from_order_data, try to match the paymentlines with the cashregisters.
```js
 if (["print"].indexOf(action) !== -1) {
                var paymentLines = order_data.statement_ids || [];
                _.each(paymentLines, function(paymentLine) {
                    var line = paymentLine;
                    // In case of local data
                    if (line.length === 3) {
                        line = line[2];
                    }
                    _.each(self.pos.cashregisters, function(cashregister) {
                        if (cashregister.journal.id === line.journal_id) {
                            if (line.amount > 0) {
                                // If it is not change
                                order.add_paymentline(cashregister);
                                order.selected_paymentline.set_amount(line.amount);
                            }
                        }
                    });
                });
            }
```

but `self.pos.cashregisters` is obsolete, no cashregisters in pos model.

**Comment by javierjcf** [comment]
I think in the function `_prepare_done_order_payment_for_pos` we should use a new key:

`payment_method_id=payment_line.payment_method_id.id`

In the frontend we should use 
```
  _.each(self.pos.payment_methods, function(cashregister) {
                        if (cashregister.id === line.payment_method_id) {
```

Instead of 
```
  _.each(self.pos.cashregisters, function(cashregister) {
                        if (cashregister.journal.id === line.journal_id) {
```

I'm not sure at all if this is the best way to do it, but the fix is working for me in a custom module.

**Comment by legalsylvain** [comment]
Could you propose a PR ?
Thanks.

**Comment by javierjcf** [comment]
PR propossed:
#733

**Comment by javierjcf** [comment]
Hi, 

This issue can be closed, its solved on #733

**Comment by chienandalu** [comment]
Sure. Thanks :)
